### ISSUE TITLE
Support API Key as environment variable

### ISSUE THREAD

**Issue by ruflin** [elastic/elastic-package#1633]
Currently if connecting elastic-package to a remote Elastic Stack, only username / password as environment variable are supported. As we are moving more and more to API Keys, also an API key should be supported. See https://github.com/elastic/elastic-package/pull/1627 for more discussions.

Stretch goal: Ideally we would have an easy way to create an API Key either through and API call or in the UI in Kibana with the right permissions ...

Related PRs:
* https://github.com/elastic/elastic-package/pull/2298
* https://github.com/elastic/elastic-package/pull/2307
* https://github.com/elastic/elastic-package/pull/2308
* https://github.com/elastic/elastic-package/pull/2309
* https://github.com/elastic/elastic-package/pull/2317
* https://github.com/elastic/elastic-package/pull/2338

**Comment by jsoriano** [comment]
There are currently issues installing packages with API keys https://github.com/elastic/kibana/issues/166473

**Comment by bhapas** [comment]
@jsoriano Is this unblocked when https://github.com/elastic/kibana/pull/178120 is merged?

**Comment by jsoriano** [comment]
> @jsoriano Is this unblocked when [elastic/kibana#178120](https://github.com/elastic/kibana/pull/178120) is merged?

Looks like that, yes.

**Comment by andrewkroh** [comment]
We need API key support for interacting with Serverless projects. Unless you create the project through the API you won't have an admin user/pass and you cannot create local users. So API key is the only option.

**Comment by jsoriano** [comment]
Updated description with related PRs.

Apart from changes similar to Andrew's (thanks!), I am adding an "environment" stack provider that can be used when elastic-package is configured with environment variables, to fill the gaps for system testing.

I have added related PRs to the description.
